kong/endpoint_permission_service_test.go

```diff
@@ -87,3 +88,56 @@ func TestRBACEndpointPermissionservice(T *testing.T) {
 	err = client.Workspaces.Delete(defaultCtx, createdWorkspace.ID)
 	require.NoError(T, err)
 }
+
+func TestRBACEndpointPermissionList(t *testing.T) {
+	RunWhenEnterprise(t, ">=0.33.0", RequiredFeatures{RBAC: true})
+
+	client, err := NewTestClient(nil, nil)
+	require.NoError(t, err)
+	assert.NotNil(t, client)
+
+	pageSize := 1000
+	workspace := &Workspace{
+		Name: String("rbac-endpoint-permission-list-endpoint-test-workspace"),
+	}
+
+	createdWorkspace, err := client.Workspaces.Create(defaultCtx, workspace)
+	require.NoError(t, err)
+	assert.NotNil(t, createdWorkspace)
+
+	workspaceClient, err := NewTestClient(
+		String(defaultBaseURL+"/rbac-endpoint-permission-list-endpoint-test-workspace"), nil)
+	require.NoError(t, err)
+
+	role, err := workspaceClient.RBACRoles.Create(defaultCtx, &RBACRole{Name: String("roleA")})
+	require.NoError(t, err)
+	assert.NotNil(t, role)
+
+	t.Run("load all pages", func(t *testing.T) {
+		for i := 0; i < pageSize+1; i++ {
```

Comment:
Creating 1001 endpoint permissions seems a little slow - I'm open to suggestions if there is a faster way of doing it.

Reply:
It should be enough if you create only 2 endpoints as the `pageSize` used in the client is set on tests startup to 1: https://github.com/Kong/go-kong/blob/f07287df3813c28cb3b97989e3a5c7333abb6203/kong/client_test.go#L169